datahub-graphql-core/src/main/resources/entity.graphql

```diff
@@ -3209,6 +3265,237 @@ input InstitutionalMemoryMetadataUpdate {
     createdAt: Long
 }
 
+"""
+A Notebook Metadata Entity
+"""
+type Notebook implements Entity {
+    """
+    The primary key of the Notebook
+    """
+    urn: String!
+
+    """
+    A standard Entity Type
+    """
+    type: EntityType!
+
+    """
+    The Notebook tool name
+    """
+    tool: String!
+
+    """
+    An id unique within the Notebook tool
+    """
+    notebookId: String!
+
+    """
+    Additional read only information about the Notebook
+    """
+    info: NotebookInfo
+
+    """
+    Additional read write properties about the Notebook
+    """
+    editableProperties: NotebookEditableProperties
+
+    """
+    Ownership metadata of the Notebook
+    """
+    ownership: Ownership
+
+    """
+    Status metadata of the Notebook
+    """
+    status: Status
+
+    """
+    The content of this Notebook
+    """
+    content: NotebookContent!
+
+    """
+    The tags associated with the Notebook
+    """
+    tags: GlobalTags
+
+    """
+    References to internal resources related to the Notebook
+    """
+    institutionalMemory: InstitutionalMemory
+
+    """
+    The Domain associated with the Notebook
+    """
+    domain: Domain
+
+    """
+    Edges extending from this entity
+    """
+    relationships(input: RelationshipsInput!): EntityRelationshipsResult
+
+    """
+    Sub Types that this entity implements
+    """
+    subTypes: SubTypes
+
+    """
+    The structured glossary terms associated with the notebook
+    """
+    glossaryTerms: GlossaryTerms
+}
+
+"""
+The actual content in a Notebook
+"""
+type NotebookContent {
+    """
+    The content of a Notebook which is composed by a list of NotebookCell
+    """
+    cells: [NotebookCell]!
```

Comment:
cell cannot be null right? should be [NotebookCell!]!

Reply:
ah, thanks for this! Not realized that I need to put ! inside the []